exports/dataset.cpp

```diff
@@ -954,6 +1066,30 @@ PYBIND11_MODULE(dataset, m) {
       // TODO Make sure we have all overloads covered to avoid implicit
       // conversion of DatasetSlice to Dataset.
       .def(py::self == py::self, py::call_guard<py::gil_scoped_release>())
+      .def(py::self != py::self, py::call_guard<py::gil_scoped_release>())
```

Comment:
These have all been added, but implicit conversion still in place?

Reply:
See above, we need to avoid unnecessary implicit conversions to minimize copies.